src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -938,150 +943,73 @@ public StringBuilder Remove(int startIndex, int length)
-        public StringBuilder Append(float value)
-        {
-            Contract.Ensures(Contract.Result<StringBuilder>() != null);
-            return Append(value.ToString());
-        }
+        public StringBuilder Append(float value) => Append(value.ToString());
 
-        // Appends a double to this string builder. 
-        // The capacity is adjusted as needed. 
-        public StringBuilder Append(double value)
-        {
-            Contract.Ensures(Contract.Result<StringBuilder>() != null);
-            return Append(value.ToString());
-        }
+        public StringBuilder Append(double value) => Append(value.ToString());
 
-        public StringBuilder Append(decimal value)
-        {
-            Contract.Ensures(Contract.Result<StringBuilder>() != null);
-            return Append(value.ToString());
-        }
+        public StringBuilder Append(decimal value) => Append(value.ToString());
 
-        // Appends an ushort to this string builder. 
-        // The capacity is adjusted as needed. 
         [CLSCompliant(false)]
-        public StringBuilder Append(ushort value)
-        {
-            Contract.Ensures(Contract.Result<StringBuilder>() != null);
-            return Append(value.ToString());
-        }
+        public StringBuilder Append(ushort value) => Append(value.ToString());
 
-        // Appends an uint to this string builder. 
-        // The capacity is adjusted as needed. 
         [CLSCompliant(false)]
-        public StringBuilder Append(uint value)
-        {
-            Contract.Ensures(Contract.Result<StringBuilder>() != null);
-            return Append(value.ToString());
-        }
+        public StringBuilder Append(uint value) => Append(value.ToString());
 
-        // Appends an unsigned long to this string builder. 
-        // The capacity is adjusted as needed. 
         [CLSCompliant(false)]
-        public StringBuilder Append(ulong value)
-        {
-            Contract.Ensures(Contract.Result<StringBuilder>() != null);
-            return Append(value.ToString());
-        }
+        public StringBuilder Append(ulong value) => Append(value.ToString());
 
-        // Appends an Object to this string builder. 
-        // The capacity is adjusted as needed. 
-        public StringBuilder Append(Object value)
-        {
-            Contract.Ensures(Contract.Result<StringBuilder>() != null);
-
-            if (null == value)
-            {
-                //Appending null is now a no-op.
-                return this;
-            }
-            return Append(value.ToString());
-        }
+        public StringBuilder Append(object value) => value == null ? this : Append(value.ToString());
 
-        // Appends all of the characters in value to the current instance.
         public StringBuilder Append(char[] value)
         {
             Contract.Ensures(Contract.Result<StringBuilder>() != null);
 
-            if (null != value && value.Length > 0)
+            if (value?.Length > 0)
```

Comment:
@stephentoub did you tell me at some point that it was more efficient if it's a hot path to use check null and length separately rather than use the null coalescing operator? I compared the IL of the two and it's not clera to me.